watchgod/watcher.py

```diff
@@ -25,7 +25,21 @@ def should_watch_dir(self, entry):
     def should_watch_file(self, entry):
         return True
 
+    def _watch_file(self, mtime, path, new_files, changes):
```

Comment:
this can be cleaned up quite a lot, something like

```py
def _walk(self, path, changes, new_files):
    if os.path.isfile(path):
        self._watch_file(...)
    else:
        self._walk_dir(...)

def _watch_file(...):
    ...

def _walk_dir(...):
    for entry in os.scandir(dir_path):
        calls either _watch_file or _walk_dir
```